docs/ref/applications.txt

```diff
@@ -56,25 +56,16 @@ application and have models, etc. (which would require adding it to
 Configuring applications
 ========================
 
-To configure an application, subclass :class:`~django.apps.AppConfig` and put
-the dotted path to that subclass in :setting:`INSTALLED_APPS`.
+To configure an application, create an ``apps.py`` module inside the
+application, then define a subclass of :class:`~django.apps.AppConfig` there.
 
 When :setting:`INSTALLED_APPS` contains the dotted path to an application
-module, Django checks for a ``default_app_config`` variable in that module.
+module, if Django finds exactly one :class:`~django.apps.AppConfig` subclass
+in the ``apps.py`` submodule, it uses that configuration for the application.
+Else, it uses the base :class:`~django.apps.AppConfig` class.
```

Comment:
What about the `default=True`? — I think we need to be super explicit here. 

(Copying from the comment on the release notes diff):

There's a three step algorithm: 

1.  Automatic discovery of single AppConfig.
2. Finer-grained control using default.
3. Explicit class path in INSTALLED_APPS.

I think we have to spell that out here. (Then perhaps a link back from the release notes... ` See ... for more details.`)